analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -0,0 +1,411 @@
+# get path to fry output directory 
+fry_dir <- file.path(s3_out, "alevin-fry-knee", sample_id)
+fry_dir <- paste0(fry_dir, "-spliced_intron_txome_k31-salign-cr-like-em-knee")
+
+# paths to spatial folders 
+cellranger_folder <- paste0(sample_id, "-cdna-spatial")
+spatial_dir <- file.path(s3_out, "cellranger", cellranger_folder, "outs", "spatial")
+```
+
+
+```{r}
+# read in combined fry and spaceranger spe 
+fry_spe <- create_fry_spaceranger_spe(fry_dir, 
+                                       spatial_dir, 
+                                       sample_id)
+```
+
+```{r}
+# spaceranger output paths
+spaceranger_out <- file.path(s3_out, "cellranger", cellranger_folder, "outs")
+
+# read in spaceranger output directly using read10XVisium
+spaceranger_spe <- create_spaceranger_spe(spaceranger_out, sample_id)
+```
+
+### Per Spot QC Metrics
+
+Now that we have read in the data and created our two `SpatialExperiment` objects, we can go ahead and combine them into one list and then calculate the per spot QC metrics using `scuttle::addPerCellQCMetrics().`
+
+```{r}
+# create one list with both spe's together
+all_spe_list <- list(fry_spe, spaceranger_spe)
+names(all_spe_list) <- c("alevin-fry", "spaceranger")
+
+# calculate per cell QC and output to a combined data frame with plotting 
+all_spe_list <- all_spe_list %>%
+    purrr::map(
+      ~ scuttle::addPerCellQCMetrics(.x, 
+                                     subsets = list(mito = mito_genes[mito_genes %in% rownames(.x)]))) %>%
+  purrr::map(scuttle::addPerFeatureQCMetrics)
+```
+
+After adding in the per spot QC metrics to both of the spe's, we want to extract the `colData` from each spe and create a data frame that we can use for plotting. 
+We will also need some information about each sample and how it was run, so we will create a sample metadata table, `sample_info_df` that will then be merged with the `colData`. 
+
+```{r}
+# create sample info dataframe to be joined with per spot dataframe later
+sample_info_df <- quant_info_table(data_dir= s3_out, 
+                 tools = c("cellranger", "alevin-fry-knee"),
+                 samples = sample_id) %>%
+  # convert cellranger to spaceranger 
+  dplyr::mutate(tool = ifelse(tool == "cellranger", "spaceranger", tool))
+```
+
+When we convert the `colData` to a data frame we use the custom function, `spatial_coldata_to_df()` to do so and apply it to each spe in our list. 
+```{r}
+# join coldata dataframe with sample info
+coldata_df <- purrr::map_df(all_spe_list, spatial_coldata_to_df, .id = "tool") %>%
+  # remove extra -1 from spaceranger barcodes
+  dplyr::mutate(spot_id = gsub("-1", "", spot_id)) %>%
+  dplyr::left_join(sample_info_df,
+                   by = c("tool"))
+```
+
+Now we only want to filter our data frame to contain spots that are shared between both tools and those that are found to be overlapping with the tissue. 
+
+```{r}
+# identify shared spots only 
+spot_counts <- coldata_df %>%  
+  dplyr::count(spot_id, sample)
```

Comment:
I think we should see what this result is? I would expect that we would get the same spots, but if we are not, how often is that happening?